NonSucking.Framework.Serialization/GeneratedSerializerCode.cs

```diff
@@ -0,0 +1,151 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using VaVare.Statements;
+
+namespace NonSucking.Framework.Serialization;
+
+public class GeneratedSerializerCode
+{
+    public List<SerializerVariable> VariableDeclarations { get; } = new();
+    public List<StatementSyntax> Statements { get; } = new();
+
+    public IEnumerable<StatementSyntax> ToMergedBlock()
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            yield return declaration.ToDeclarationAndAssignment();
+        }
+
+        foreach (var statement in Statements)
+        {
+            yield return statement;
+        }
+    }
+
+    public IEnumerable<StatementSyntax> MergeBlocksSeperated(GeneratedSerializerCode other,
+        Func<SerializerVariable, LocalDeclarationStatementSyntax> variableTransformer = null)
+    {
+        foreach (var declaration in VariableDeclarations)
+        {
+            var newDeclaration = variableTransformer == null
+                                    ? declaration.Declaration
+                                    : variableTransformer(declaration);
+            
+            other.VariableDeclarations.Add(new SerializerVariable(newDeclaration, declaration.OriginalMember, declaration.UniqueName, null));
+        }
+        return VariableDeclarations.Where(x => x.InitialValue != null).Select(x => x.GetAssignment()).Concat(Statements);
```

Comment:
The same here. Is not null should be possible, because we are on a higher language version
```suggestion
        return VariableDeclarations.Where(x => x.InitialValue is not null).Select(x => x.GetAssignment()).Concat(Statements);
```